packages/core/src/StateNode.ts

```diff
@@ -972,23 +972,24 @@ class StateNode<
   }
 
   private getExternalReentryNodes(
-    childStateNode: StateNode<TContext, any, TEvent, any, any, any>
+    targetNode: StateNode<TContext, any, TEvent, any, any, any>
   ): Array<StateNode<TContext, any, TEvent, any, any, any>> {
     const nodes: Array<StateNode<TContext, any, TEvent, any, any, any>> = [];
-    let marker:
-      | StateNode<TContext, any, TEvent, any, any, any>
-      | undefined = childStateNode;
+    let [marker, possibleAncestor]: [
+      StateNode<TContext, any, TEvent, any, any, any> | undefined,
+      StateNode<TContext, any, TEvent, any, any, any>
+    ] = targetNode.order > this.order ? [targetNode, this] : [this, targetNode];
 
-    while (marker && marker !== this) {
+    while (marker && marker !== possibleAncestor) {
       nodes.push(marker);
       marker = marker.parent;
     }
-    if (marker !== this) {
-      // we never got to `this`, therefore the `childStateNode` "escapes" it
+    if (marker !== possibleAncestor) {
+      // we never got to `possibleAncestor`, therefore the initial `marker` "escapes" it
       // it's in a different part of the tree so no states will be reentered for such an external transition
       return [];
     }
-    nodes.push(this);
+    nodes.push(possibleAncestor);
```

Comment:
Observation - this sort of uses the parent of the found ancestor as the transition domain which means that we are using LCA algorithm here whereas SCXML is using LCCA. I'm not exactly sure why SCXML has switched to LCCA because I think that they have been using LCA at the beginning. It feels to me that LCA is an easier rule and LCCA is a little bit harder to explain so I wonder what made them switch those algorithms.

Reply:
Would this break any SCXML tests?

Reply:
Using LCA over LCCA? Sure. Using LCCA can lead to [some weird behaviors](https://lists.w3.org/Archives/Public/www-voice/2013JanMar/0059.html) when parallel states are involved though and I'm not sure why those weird behaviors would be desirable in the first place. We should dig into this more to understand the motivation for using LCCA in UML.